workspaces/ui-v2/src/optic-components/hooks/config/AppConfiguration.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { ReactNode, useContext } from 'react';
+
+interface IFeatureFlags {}
+
+interface IAppConfigurations {
+  navigation: {
+    showDocs?: boolean;
+    showChangelog?: boolean;
+    showDiff?: boolean;
+  };
+  documentation: {
+    allowDescriptionEditing: boolean;
+  };
+}
+
+export type OpticAppConfig = {
+  featureFlags: IFeatureFlags;
+  config: IAppConfigurations;
+};
+
+const AppConfigurationContext = React.createContext<OpticAppConfig | null>(
+  null
+);
+
+export const AppConfigurationStore = (props: {
+  config: OpticAppConfig;
+  children: ReactNode;
+}) => {
+  return (
+    <AppConfigurationContext.Provider value={props.config}>
+      {props.children}
+    </AppConfigurationContext.Provider>
+  );
+};
+
+export function useFeatureFlags() {
+  return useContext(AppConfigurationContext)?.featureFlags!;
```

Comment:
seems like we have some extra punctuation in a lot of places (for example the ! here), not sure where to start auditing that, but maybe walk through the ?s and !s (not just in this changeset) and the related types to make sure they can be stated more definitively

Reply:
seems like it should be useContext(X)! if the assumption is that it will always be there

Reply:
One way that we could handle this is do something like...
```
export function useFeatureFlags() {
    const value = useContext(AppConfigurationContext)
    if (!value) { throw new Error("useFeatureFlags could not find AppConfigurationContext")
   return value
}
```

This way value is asserted that we have context, as well as if this is misused, the error is caught much quicker in the development cycle (rather than useFeatureFlags returning undefined instead of a boolean)

Reply:
Done @niclim  -- but I subbed for `invariant` lib

Reply:
That works - the only other consideration here is typescript can infer from value + error thrown blocks but can't infer through functions (i.e. you still need the `!` punctuation with `invariant` but dont need it with a if -> throw error block)